Tetris_formal.py

```diff
@@ -19,57 +23,97 @@
 
 
-######### 記錄各種方塊的上下左右
-class T_block():
-    
+class InitializeData:
+    """
+    各個方塊的初始條件
+    """
+
     def __init__(self, x, y):
         self.x = x
         self.y = y
-        
-    def get_direction(self, color):
-    
 
-        up = [[color, self.x, self.y], [color, self.x + 40, self.y], [color, self.x - 40, self.y], [color, self.x, self.y - 40]]
-        down = [[color, self.x, self.y], [color, self.x + 40, self.y], [color, self.x - 40, self.y], [color, self.x, self.y + 40]]
-        right = [[color, self.x, self.y], [color, self.x, self.y - 40], [color, self.x, self.y + 40], [color, self.x + 40, self.y]]
-        left = [[color, self.x, self.y], [color, self.x, self.y - 40], [color, self.x, self.y + 40], [color, self.x - 40, self.y]]
+    def color_or_picture(self, parameter, picture_root):
+        """
+        50%的機率選定會出現照片還是顏色，其中照片和顏色都是在其他方塊的class裡匯入的
+        """
+
+        preference = random.choice(["color", "picture"])
+
+        if preference == "color":
+            color = parameter
+
+        elif preference == "picture":
+            color = pygame.image.load(picture_root)
+            color = pygame.transform.scale(color, (40, 40))
+            color.blit(color, (self.x, self.y))
+
+        return color
+
+
+class TBlock(InitializeData):
+    """
+    上下左右的紀錄是為了之後旋轉時，能辨別它的上下左右的各個方塊的所在位置的座標是什麼。
+    """
+
+    def get_direction(self, color):
+        """
+        同上
+        """
+        up = [
+            [color, self.x, self.y],
+            [color, self.x + 40, self.y],
+            [color, self.x - 40, self.y],
+            [color, self.x, self.y - 40],
+        ]
+        down = [
+            [color, self.x, self.y],
+            [color, self.x + 40, self.y],
+            [color, self.x - 40, self.y],
+            [color, self.x, self.y + 40],
+        ]
+        right = [
+            [color, self.x, self.y],
+            [color, self.x, self.y - 40],
+            [color, self.x, self.y + 40],
+            [color, self.x + 40, self.y],
+        ]
+        left = [
+            [color, self.x, self.y],
+            [color, self.x, self.y - 40],
+            [color, self.x, self.y + 40],
+            [color, self.x - 40, self.y],
+        ]
 
         return up, down, right, left
 
-    
     def get_color(self):
-        num = random.randint(1,2)
-        
-        if num == 1 :
-            color = (100,90,200)
-        
-        elif num == 2 :
-            
-            color = pygame.image.load('/Users/coolguy/Documents/picture_for_T.jpg')
-            color = pygame.transform.scale(color,(40,40))
-            color.blit(color,(self.x, self.y))
-            
-       
+        """
+        同上
+        """
+        color = self.color_or_picture(
+            (100, 90, 200), "/Users/coolguy/Documents/picture_for_T.jpg"
+        )
+
         return color
-    
 
 
-class L_block():
-    
-    def __init__(self, x, y):
-        self.x = x
-        self.y = y
-    
-    
+class LBlock(InitializeData):
+    """
```

Comment:
Ditto. The description doesn't mean anything. Please briefly describe the function.